superflore/generators/bitbake/run.py

```diff
@@ -71,14 +71,10 @@ def main():
             err('Failed to file PR!')
             err('reason: {0}'.format(e))
             sys.exit(1)
-    elif args.ros_distro:
-        warn('"{0}" distro detected...'.format(args.ros_distro))
-        selected_targets = [args.ros_distro]
-        preserve_existing = False
-    elif args.only:
-        parser.error('Invalid args! --only requires specifying --ros-distro')
-    if not selected_targets:
-        selected_targets = get_distros_by_status('active')
+    warn('"{0}" distro detected...'.format(args.ros_distro))
+    selected_targets = [args.ros_distro]
+    preserve_existing = False
```

Comment:
`preserve_existing` should remain `True` if `args.only` is `True` => drop line 44 and replace 76 with: `preserve_existing = args.only`